src/lap_stats/lap_stats.c

```diff
@@ -82,15 +82,210 @@ static float g_distance;
+         * line (logical because the last sector boundary is always the finish
+         * line).  So we must compare sector 0 against the finish line and if
+         * they are the same, then we do not have sector values defined.
+         */
+        const GeoPoint sp0 = getSectorGeoPointAtIndex(track, 0);
+        const GeoPoint fin = getFinishPoint(track);
+        return !are_geo_points_equal(&fin, &sp0);
+}
+
+
+/**
+ * Sets the active track that lap_stats uses for determining its various
+ * statistics.
+ * @param track Pointer to the track where we are at.
+ * @param radius Radius of the geo circles at this track in meters.
+ * @param track_status Status indicating who set the track value.
+ * @return true if setup was successful, false otherwise.
+ */
+static bool _set_active_track(const Track *track, const float radius,
+                              const track_status_t track_status)
+{
+        /* We are changing our track, so we need to reset stats */
+        lapstats_reset();
+        reset_track();
+        g_track_status = track_status;
+        g_configured = 1;
+
+        if (!track || !radius)
+                return false;
+
+        memcpy(&g_active_track, track, sizeof(Track));
+        g_geo_circle_radius = radius;
+
+        g_start_finish_enabled = isStartFinishEnabled(track);
+        g_sector_enabled = isSectorTrackingEnabled(track);
+
+        setup_geo_triggers(track, radius * GEO_TRIGGER_RADIUS_MULTIPLIER);
+        setup_geo_circles(track, radius);
+        lc_setup(track, radius);
+
+        return true;
+}
+
+bool lapstats_set_active_track(const Track *track, const float radius)
 {
-        g_active_track = t;
+        return _set_active_track(track, radius, TRACK_STATUS_EXTERNALLY_SET);
 }
 
-static float degrees_to_meters(float degrees)
+/**
+ * Basic method that calculates distances represented in degrees of latitude
+ * at the equator into meters.  Only present to deal with legacy issues.
+ */
+float lapstats_degrees_to_meters(const float degrees)
 {
-    // There are 110574.27 meters per degree of latitude at the equator.
-    return degrees * 110574.27;
+        /*
+         * LEGACY HACK. Our original Track object had the radius of
+         * geoCircle measured in degrees!!! This method exists to convert
+         * from that measurement to one that is measured in meters, a more
+         * civilized unit of measure. I would love to murder the need to use
+         * this in the future.
+         * 110574.27 meters per degree of latitude at the equator.
+         */
+        return degrees * 110574.27;
+}
+
+/**
+ * Informs the caller as to whether or not the currenctly configured
+ * track (if one is configured at all) is valid. By valid we mean that
+ * the track is capable of recording laps (valid start/finish points
+ * are defined).
+ * @return true if valid start/finish points exist, false otherwise.
+ */
+bool lapstats_is_track_valid()
```

Comment:
We already do check for it when the `set_active_track` method is invoked.  I upgraded it so that it enforces a minimum radius of 1 meter.